### PR TITLE
feat(drive): add +update-title shortcut

### PR DESCRIPTION
**Pull request by zzzchao123**
## Summary

Wrap the Drive title endpoint as drive +update-title. Accept a resource URL (recommended) or bare token + --type for the target, covering every type the endpoint takes: docx, doc, sheet, bitable (base alias), slides, mindnote, file, folder and wiki. --title carries the new title and also accepts --new-title, matching the API field name.

Wiki inputs are patched as the wiki node they name: the endpoint resolves type=wiki against the node token itself, so a /wiki/ URL needs no get_node unwrapping step, and node and underlying document titles stay in sync either way it is renamed. A whitespace-only title is rejected locally because the server accepts an empty new_title and leaves the resource with a blank title. The two codes this endpoint collapses many causes into get command-level hints: 981003 explains that the lookup runs under the declared type (and, for wiki, that the document token is not accepted there), 981004 points at the missing edit permission.

Pre-flight scopes are intentionally empty: the endpoint's scope set is an any-of that depends on --type (docx:document:write_only, sheets:spreadsheet:write_only, base:app:update, drive:file:upload), and unconditional pre-flight is ALL-of, so declaring them there would reject tokens holding only the scope for the target type. They are declared as conditional scopes instead, and the per-type mapping is documented in the skill reference.

Cover the shortcut with unit tests, dry-run e2e and a self-contained live workflow (upload -> rename file -> verify -> rename folder -> verify -> type-mismatch negative leg), and register it in
tests/cli_e2e/drive/coverage.md. Route rename intents in the lark-drive skill to the shortcut instead of the raw files patch service command.

## Changes

- Change 1
- Change 2

## Test Plan

- [x] Unit tests pass
- [x] Manual local verification confirms the `lark-cli <domain> <command>` flow works as expected

## Related Issues

- None

## Summary by CodeRabbit

* **New Features**
  * Added `drive +update-title` for renaming files, folders, documents, bitables, slides, mindnotes, and wiki nodes.
  * Supports URLs or tokens, type validation, title trimming, dry-run mode, structured output, and API error guidance.
  * Added configurable file-extension handling, including preservation, rejection, or explicit override.

* **Documentation**
  * Added usage instructions, supported resource types, URL and token rules, extension policies, and examples.

* **Tests**
  * Added comprehensive unit and end-to-end coverage for validation, dry runs, workflows, extension handling, and error scenarios.

### CODE REVIEW

File: skills/lark-drive/references/lark-drive-update-title.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+|------|------|------|
+| `--url` | 与 `--token` 二选一 | 目标 URL，支持 `/docx/`、`/doc/`、`/sheets/`、`/base/`、`/bitable/`、`/slides/`、`/mindnote/`、`/file/`、`/drive/folder/`、`/wiki/` |
+| `--token` | 与 `--url` 二选一 | 目标 token 或 URL；裸 token 必须配合 `--type` |
+| `--type` | 裸 token 时必填 | `docx`、`doc`、`sheet`、`bitable`（`base` 为兼容别名）、`slides`、`mindnote`、`file`、`folder`、`wiki`；传 URL 时可省略，显式传入时必须与 URL 类型一致 |
+| `--title` | 是 | 新标题，别名 `--new-title`；不能为空或纯空白 |
+
+## 行为说明
+
+- **空标题会清空标题**：服务端接受空 `new_title` 并把目标标题置空，所以 CLI 本地就拒绝空或纯空白的 `--title`
+- **`file` 类型连扩展名一起替换**：`--type file` 的标题就是完整文件名，扩展名（`.xlsx`、`.txt` 等）不写进 `--title` 就会丢掉，后续按扩展名分流的下载、预览、导入都会受影响
```
Comment (wittam-01) [inline skills/lark-drive/references/lark-drive-update-title.md]: file类型，看是否要做下前置校验，如果新标题和原始标题后缀，新标题如果没有后缀或者后缀不一致，返回报错之类的，如把1.md改成2.txt或者2，然后看是否要加个额外参数可以控制不校验。  可以ai分析下怎么处理比较合适，不然用户改错了就没了

File: skills/lark-drive/references/lark-drive-update-title.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+## 参数
+
+| 参数 | 必填 | 说明 |
+|------|------|------|
+| `--url` | 与 `--token` 二选一 | 目标 URL，支持 `/docx/`、`/doc/`、`/sheets/`、`/base/`、`/bitable/`、`/slides/`、`/mindnote/`、`/file/`、`/drive/folder/`、`/wiki/` |
+| `--token` | 与 `--url` 二选一 | 目标 token 或 URL；裸 token 必须配合 `--type` |
+| `--type` | 裸 token 时必填 | `docx`、`doc`、`sheet`、`bitable`（`base` 为兼容别名）、`slides`、`mindnote`、`file`、`folder`、`wiki`；传 URL 时可省略，显式传入时必须与 URL 类型一致 |
+| `--title` | 是 | 新标题，别名 `--new-title`；不能为空或纯空白 |
+
+## 行为说明
```
Comment (wittam-01) [inline skills/lark-drive/references/lark-drive-update-title.md:37]: 结合错误监控补充场景错误引导

File: tests/cli_e2e/drive/coverage.md [hunk 3]
```diff
@@ -1,9 +1,9 @@
 # Drive CLI E2E Coverage
 
 ## Metrics
-- Denominator: 40 leaf commands
-- Covered: 22
-- Coverage: 55.0%
+- Denominator: 41 leaf commands
+- Covered: 23
+- Coverage: 56.1%
```
[the remark this answers is not in the document]
Comment (zzzchao123) [reply tests/cli_e2e/drive/coverage.md]: 已修复

File: skills/lark-drive/references/lark-drive-update-title.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+```json
+{
+  "updated": true,
+  "file_token": "<file_token>",
+  "type": "docx",
+  "title": "<new_title>",
+  "url": "https://example.feishu.cn/docx/<file_token>"
+}
+```
```
[the remark this answers is not in the document]
Comment (zzzchao123) [reply skills/lark-drive/references/lark-drive-update-title.md:55]: 无需修改

File: skills/lark-drive/references/lark-drive-update-title.md (hunk 1 above)
Comment (wittam-01) [reply skills/lark-drive/references/lark-drive-update-title.md]: 2 自动补充后缀

Comment (zzzchao123) [reply skills/lark-drive/references/lark-drive-update-title.md]: 默认走「自动补后缀」：

--type file 时先读一次当前文件名（metas.batch_query），然后按新增的 --on-extension-mismatch 处理，默认 keep：

标题没有后缀 → 自动补上当前后缀，输出里用 extension_appended 说明（1.md + 标题 2 → 2.md）
标题改了后缀（1.md → 2.txt）→ 默认报错，不自动纠正：这是明确的相反意图，替用户改回去等于跟他对着干
--on-extension-mismatch=error：缺后缀也报错，一律不修
--on-extension-mismatch=allow：原样提交（确认要去掉/换后缀时用），同时跳过那次读取，所以没有 drive:drive.metadata:readonly 的调用方也能改名

File: skills/lark-drive/references/lark-drive-update-title.md (hunk 2 above)
Comment (zzzchao123) [reply skills/lark-drive/references/lark-drive-update-title.md:37]: 已添加限频和scope两种错误码引导

File: skills/lark-drive/SKILL.md [hunk 5]
```diff
@@ -52,7 +52,7 @@ metadata:
 - `drive +inspect` / `drive +upload` 遇到 `not found`、`permission denied`、`missing scope` 时，默认停止重试；只有 `rate limit` 或临时网络错误才适合有限重试。
 
 ## 修改标题
-- 使用 `drive files patch` 命令，通过new_title字段可以修改标题，支持 docx、sheet、bitable、file、wiki、folder 类型
+- 用户要**重命名 / 改标题 / 改文件名**，使用 `lark-cli drive +update-title`，用法见 [`references/lark-drive-update-title.md`](references/lark-drive-update-title.md)。
```
[the remark this answers is not in the document]
Comment (zzzchao123) [reply skills/lark-drive/SKILL.md:55]: 已修复

File: shortcuts/drive/drive_update_title.go [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+	return spec, nil
+}
+
+// validateDriveUpdateTitleSpec keeps the CLI from issuing a request that the
+// server accepts but that leaves the resource without a title.
+func validateDriveUpdateTitleSpec(spec driveUpdateTitleSpec) error {
+	if spec.Title == "" {
+		return errs.NewValidationError(errs.SubtypeInvalidArgument, "--title must not be empty or whitespace-only").
+			WithParam("--title").
+			WithHint("the API accepts an empty new_title and clears the resource title; pass the exact title you want to see")
```
[the remark this answers is not in the document]
Comment (zzzchao123) [reply shortcuts/drive/drive_update_title.go:264]: 已修复

File: skills/lark-drive/references/lark-drive-update-title.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+|------|------|------|
+| `--url` | 与 `--token` 二选一 | 目标 URL，支持 `/docx/`、`/doc/`、`/sheets/`、`/base/`、`/bitable/`、`/slides/`、`/mindnote/`、`/file/`、`/drive/folder/`、`/wiki/` |
+| `--token` | 与 `--url` 二选一 | 目标 token 或 URL；裸 token 必须配合 `--type` |
+| `--type` | 裸 token 时必填 | `docx`、`doc`、`sheet`、`bitable`（`base` 为兼容别名）、`slides`、`mindnote`、`file`、`folder`、`wiki`；传 URL 时可省略，显式传入时必须与 URL 类型一致 |
+| `--title` | 是 | 新标题，别名 `--new-title`；不能为空或纯空白。原样提交，首尾空格服务端会保留，不想要就别传 |
+| `--on-extension-mismatch` | 否 | 仅 `--type file`：`keep`（默认，标题缺后缀时自动补上当前后缀）/ `error`（后缀有任何变化就报错）/ `allow`（原样提交，且不读当前文件名）。传给其他 `--type` 会报错 |
+
+## 行为说明
+
+- **空标题会清空标题**：服务端接受空 `new_title` 并把目标标题置空，所以 CLI 本地就拒绝空或纯空白的 `--title`
```
Comment (wittam-01) [inline skills/lark-drive/references/lark-drive-update-title.md]: 简单写就好了，不需要写服务端限制，就写cli会拒绝空标题就好

File: skills/lark-drive/references/lark-drive-update-title.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+- **`file` 类型连扩展名一起替换**：`--type file` 的标题就是完整文件名，扩展名不写进 `--title` 服务端就会把它丢掉，后续按扩展名分流的下载、预览、导入都会受影响
```
Comment (wittam-01) [inline skills/lark-drive/references/lark-drive-update-title.md]: 这两句也是，不需要暴露cli内部实现细节（依赖api和scope），简单说明内部会校验title后缀和文件当前标题后缀是否一致即可，没有后缀会默认补充后缀，如果不一样会拦截，如果需要跳过拦截，需要加上xx就可以

File: skills/lark-drive/references/lark-drive-update-title.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+- **默认会兜住丢后缀**：`--type file` 时 CLI 先读一次当前文件名（`metas.batch_query`，需要 `drive:drive.metadata:readonly`），标题缺后缀就补上当前后缀并在输出的 `extension_appended` 里说明；标题改了后缀（`a.md` → `a.txt`）默认报错，确认要改就加 `--on-extension-mismatch=allow`。`allow` 同时跳过这次读取，没有 metadata scope 时也能改名
+- **wiki 不解包**：`--type wiki` 用 `/wiki/` URL 里的 `wiki_token`，传底层文档 token 会 `981003`
+- **不支持妙搭 apps**：`/page/` 资源不在端点支持范围（服务端按字段校验直接拒 `type=apps`）；改妙搭应用名用 `lark-cli apps +update --app-id <app_id> --name '<新名称>'`，传 apps URL/type 时 CLI 会用报错 hint 给出这条命令
```
Comment (wittam-01) [inline skills/lark-drive/references/lark-drive-update-title.md]: 这种跨其他业务域了，在这里引导去miaoda的skill读取就好，如：
不支持miaoda apps：如果要修改标题，切换到lark-apps业务域处理。

跨其他业务域不要有太多细节，不然后续他们改了我们也需要同步改，代码直接拒绝掉就好

File: skills/lark-drive/references/lark-drive-update-title.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+## 参数
+
+| 参数 | 必填 | 说明 |
+|------|------|------|
+| `--url` | 与 `--token` 二选一 | 目标 URL，支持 `/docx/`、`/doc/`、`/sheets/`、`/base/`、`/bitable/`、`/slides/`、`/mindnote/`、`/file/`、`/drive/folder/`、`/wiki/` |
+| `--token` | 与 `--url` 二选一 | 目标 token 或 URL；裸 token 必须配合 `--type` |
+| `--type` | 裸 token 时必填 | `docx`、`doc`、`sheet`、`bitable`（`base` 为兼容别名）、`slides`、`mindnote`、`file`、`folder`、`wiki`；传 URL 时可省略，显式传入时必须与 URL 类型一致 |
+| `--title` | 是 | 新标题，别名 `--new-title`；不能为空或纯空白。原样提交，首尾空格服务端会保留，不想要就别传 |
+| `--on-extension-mismatch` | 否 | 仅 `--type file`：`keep`（默认，标题缺后缀时自动补上当前后缀）/ `error`（后缀有任何变化就报错）/ `allow`（原样提交，且不读当前文件名）。传给其他 `--type` 会报错 |
```
Comment (wittam-01) [inline skills/lark-drive/references/lark-drive-update-title.md]: 如果是keep的情况下，传了后缀但是不匹配呢？

Comment (wittam-01) [reply skills/lark-drive/references/lark-drive-update-title.md]: 建议使用去掉error，保留keep默认（补充后缀 + 强校验），allow：只有需要改变文件后缀不一样才传，风险是修改后文件可能不可用

File: skills/lark-drive/references/lark-drive-update-title.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+```
+
+## 参数
+
+| 参数 | 必填 | 说明 |
+|------|------|------|
+| `--url` | 与 `--token` 二选一 | 目标 URL，支持 `/docx/`、`/doc/`、`/sheets/`、`/base/`、`/bitable/`、`/slides/`、`/mindnote/`、`/file/`、`/drive/folder/`、`/wiki/` |
+| `--token` | 与 `--url` 二选一 | 目标 token 或 URL；裸 token 必须配合 `--type` |
+| `--type` | 裸 token 时必填 | `docx`、`doc`、`sheet`、`bitable`（`base` 为兼容别名）、`slides`、`mindnote`、`file`、`folder`、`wiki`；传 URL 时可省略，显式传入时必须与 URL 类型一致 |
+| `--title` | 是 | 新标题，别名 `--new-title`；不能为空或纯空白。原样提交，首尾空格服务端会保留，不想要就别传 |
```
Comment (wittam-01) [inline skills/lark-drive/references/lark-drive-update-title.md]: 原样提交，首尾空格服务端会保留，不想要就别传 --这个去掉吧，加下去掉首尾空格

File: skills/lark-drive/references/lark-drive-update-title.md [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+
+| 错误码 | 含义 | 处理 |
+|---|---|---|
+| `99991672` / `99991679` | 缺失 scope | 按错误里的 `missing_scopes`、`hint` 申请/授权所需 scope 后重试 |
+| `99991400` | 命中接口限频 | 等待一段时间后重试；批量改名时保持串行并降低频率 |
+
+## 参考
+
+- [lark-drive](../SKILL.md) -- 云空间（云盘/云存储）全部命令
+- [+move](lark-drive-move.md) -- 改位置而不是改名字；知识库层级移动走 [lark-wiki](../../lark-wiki/SKILL.md)
```
Comment (wittam-01) [inline skills/lark-drive/references/lark-drive-update-title.md]: 什么情况下agent会走到这里吗，为什么需要在这里引导改位置？

File: skills/lark-drive/references/lark-drive-update-title.md [hunk 13]
```diff
@@ -0,0 +1,79 @@
+# drive +update-title
+
+> **前置条件：** 先阅读 [`../lark-shared/SKILL.md`](../../lark-shared/SKILL.md) 了解认证、全局参数和安全规则。
+
+重命名云空间（云盘/云存储）里的文件、文件夹、在线文档或知识库节点。只改标题，不改内容、位置和权限。推荐直接传 URL。
```
Comment (wittam-01) [inline skills/lark-drive/references/lark-drive-update-title.md]: 不改内容、位置和权限 --为什么需要加这个引导，会存在agent改内容/位置/权限走到这里的情况吗，看起来有点奇怪

File: shortcuts/drive/drive_update_title.go [hunk 14]
```diff
@@ -0,0 +1,530 @@
...
+	// Unconditional pre-flight is ALL-of, so declaring them there would reject
+	// tokens that legitimately hold only the scope for the target type.
+	Scopes: []string{},
+	ConditionalScopes: []string{
+		"docx:document:write_only",      // --type docx / doc
+		"sheets:spreadsheet:write_only", // --type sheet
+		"base:app:update",               // --type bitable (base)
+		"drive:file:upload",             // --type file / folder
+		driveMetadataReadScope,          // --type file extension guard reads the current title
+	},
```
Comment (wittam-01) [inline shortcuts/drive/drive_update_title.go:87]: 这个在执行lark-cli auth login的时候，选择drive业务域，会触发申请这几个scope吗？

Comment (zzzchao123) [reply shortcuts/drive/drive_update_title.go:87]: 都会申请到

Comment (wittam-01) [reply shortcuts/drive/drive_update_title.go:87]: 这里改成只保留 drive:file:upload 吧，这里的考虑是：drive:file:upload 这个其他drive的shortcut也有使用到；其他如base:app:update这种，如果没用过base的大概率没有申请到scope，如果这个scope需要审批，会造成一些额外流程。在openapi设计是只要满足任意一个scope就能调通的。

File: skills/lark-drive/references/lark-drive-update-title.md [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+
+## 参数
+
+| 参数 | 必填 | 说明 |
+|------|------|------|
+| `--url` | 与 `--token` 二选一 | 目标 URL，支持 `/docx/`、`/doc/`、`/sheets/`、`/base/`、`/bitable/`、`/slides/`、`/mindnote/`、`/file/`、`/drive/folder/`、`/wiki/` |
+| `--token` | 与 `--url` 二选一 | 目标 token 或 URL；裸 token 必须配合 `--type` |
+| `--type` | 裸 token 时必填 | `docx`、`doc`、`sheet`、`bitable`（`base` 为兼容别名）、`slides`、`mindnote`、`file`、`folder`、`wiki`；传 URL 时可省略，显式传入时必须与 URL 类型一致 |
+| `--title` | 是 | 新标题，别名 `--new-title`；不能为空或纯空白，首尾空格会被去掉 |
+| `--on-extension-mismatch` | 否 | 仅 `--type file`：`keep`（默认，标题缺后缀时自动补上当前后缀）/ `error`（后缀有任何变化就报错）/ `allow`（跳过校验，原样提交）。传给其他 `--type` 会报错 |
```
Comment (wittam-01) [inline skills/lark-drive/references/lark-drive-update-title.md]: 这个是不是只保留keep和allow。keep就是自动补后缀+如果有后缀校验是否一致，不一致报错。不然agent还需要判断当前title是否有后缀

File: shortcuts/drive/drive_update_title.go [hunk 16]
```diff
@@ -0,0 +1,529 @@
...
+			fmt.Sprintf("--title %q changes the extension from %s to %s (current name %q)", spec.Title, currentExt, newExt, currentTitle))
+	}
+	return guard, nil
+}
+
+// driveUpdateTitleExtension reads the trailing .xxx token of a file name, so the
+// guard compares exactly what the platform shows as the file's extension. A name
+// whose only dot is the leading one (".gitignore") is a dotfile, not an
+// extension, so appending it to another title would be nonsense.
+func driveUpdateTitleExtension(title string) string {
```
Comment (wittam-01) [inline shortcuts/drive/drive_update_title.go:202]: 这里如果后缀是大写，会被改成小写后才补充？和原来不一致

Comment (zzzchao123) [reply shortcuts/drive/drive_update_title.go:202]: 已修复

File: shortcuts/drive/drive_update_title.go [hunk 17]
```diff
@@ -0,0 +1,566 @@
...
+		return guard, nil
+	}
+
+	currentTitle, err := common.FetchDriveMetaTitle(runtime, spec.Ref.Token, "file")
+	if err != nil {
+		// The requested policy cannot be honored without the current name, so
+		// report that instead of renaming under a guard that never ran.
+		return driveUpdateTitleGuard{}, decorateDriveUpdateTitleCurrentTitleError(err, spec)
+	}
+	guard.PreviousTitle = currentTitle
```
Comment (fangshuyu-768) [inline shortcuts/drive/drive_update_title.go:179]: **[P2] Fail closed when metadata returns no current title**

`FetchDriveMetaTitle` returns `("", nil)` when `metas` is empty or the first meta omits `title`. Here that makes `currentExt == ""`, so the default `keep` policy continues to PATCH the user's extensionless title even though the guard never learned the current name. Treat an empty current title as an invalid response/precondition and abort before PATCH; add regression tests for both empty response shapes.

File: shortcuts/drive/drive_update_title.go [hunk 18]
```diff
@@ -0,0 +1,566 @@
...
+		if !driveUpdateTitleTypeSupported(refType) {
+			return driveUpdateTitleRef{}, errs.NewValidationError(
+				errs.SubtypeInvalidArgument,
+				"unsupported %s resource type %q; drive +update-title supports %s",
+				sourceFlag,
+				refType,
+				strings.Join(driveUpdateTitleAPITypes, ", "),
+			).WithParam(sourceFlag)
+		}
+		return driveUpdateTitleRef{Token: ref.Token, Type: refType, SourceFlag: sourceFlag}, nil
```
Comment (fangshuyu-768) [inline shortcuts/drive/drive_update_title.go:344]: **[P2] Validate the token extracted from resource URLs**

The bare-token path calls `validate.ResourceName`, but URL-derived tokens return here unchecked. `--url https://example.larksuite.com/docx/..` therefore succeeds and plans `PATCH /open-apis/drive/v1/files/..`, because `url.PathEscape` leaves `..` unchanged. Run the same `ResourceName` validation on `ref.Token` before returning the ref, preserve its cause, and add a URL regression test.

File: shortcuts/drive/drive_update_title.go [hunk 19]
```diff
@@ -0,0 +1,566 @@
...
+			Desc(fmt.Sprintf("[1] Read the current name for the --on-extension-mismatch=%s guard", spec.ExtPolicy)).
+			Body(map[string]interface{}{
+				"request_docs": []map[string]interface{}{
+					{"doc_token": spec.Ref.Token, "doc_type": "file"},
+				},
+			}).
+			PATCH("/open-apis/drive/v1/files/:file_token").
+			Desc("[2] Update the title; keep appends the extension resolved in step 1 when the title has none").
+			Params(buildDriveUpdateTitleParams(spec)).
+			Body(buildDriveUpdateTitleBody(spec)).
```
Comment (fangshuyu-768) [inline shortcuts/drive/drive_update_title.go]: **[P2] Mark the response-dependent title in the dry-run plan**

For `--type file --title renamed` with the default `keep` policy, dry-run advertises `new_title: "renamed"`, while `Execute` may read `report.xlsx` and actually send `"renamed.xlsx"` or abort on an extension mismatch. This makes the preview a plausible but false request. Use an explicit response-derived placeholder or conditional value for the second call, as adjacent multi-step dry-runs do, and assert it in the E2E test.

File: shortcuts/drive/drive_update_title.go (hunk 17 above)
Comment (zzzchao123) [reply shortcuts/drive/drive_update_title.go:179]: 已修复

File: shortcuts/drive/drive_update_title.go (hunk 18 above)
Comment (zzzchao123) [reply shortcuts/drive/drive_update_title.go:344]: 已修复

File: shortcuts/drive/drive_update_title.go (hunk 19 above)
Comment (zzzchao123) [reply shortcuts/drive/drive_update_title.go]: 已修复

**Review by fangshuyu-768 — APPROVED** [review]

**Review by fangshuyu-768 — APPROVED** [review]

**Review by liangshuo-1 — APPROVED** [review]
